crates/core/app/src/server/events.rs

```diff
@@ -38,14 +38,27 @@ impl EventIndexLayer {
                 // Perform matching on a nested key in the same format used by
                 // the cosmos SDK: https://docs.cosmos.network/main/core/config
                 // e.g., "message.sender", "message.recipient"
-                let nested_key = format!("{}.{}", e.kind, attr.key);
 
-                if self.no_index.is_match(&nested_key) {
-                    attr.index = false;
-                }
-                // This comes second so that explicit index requests take priority over no-index requests.
-                if self.index.is_match(&nested_key) {
-                    attr.index = true;
+                // XXX: tendermint-rs now supports v034 events types, which are
+                //      plain bytes and need not be utf8. This messes with the
+                //      regex and is likely not desirable to have in downstream
+                //      consumers of the indexed events.
+                match attr.key_str() {
+                    Ok(key) => {
+                        let nested_key = format!("{}.{}", e.kind, key);
+
+                        if self.no_index.is_match(&nested_key) {
+                            attr.set_index(false)
+                        }
+                        // This comes second so that explicit index requests take priority over no-index requests.
+                        if self.index.is_match(&nested_key) {
+                            attr.set_index(false)
+                        }
+                    }
+                    // TODO: what should be done with this error? Emit a warning?
+                    Err(_err) => {
+                        attr.set_index(false);
+                    }
```

Comment:
Maybe we could log here, but otherwise seems fine to just not index non-utf8 attributes. We don't generate any of those, and the basic cometbft events don't contain such keys either.